### PR TITLE
Don't evict whole cache for large entries

### LINKED ISSUE #37: Skip cache if value too big
```julia
julia> lru = LRU{Int, Vector{Int}}(; maxsize=10, by=length)
LRU{Int64, Vector{Int64}}(; maxsize = 10)

julia> get!(lru, 1, 1:9)
1:9

julia> lru
LRU{Int64, Vector{Int64}} with 1 entry:
  1 => [1, 2, 3, 4, 5, 6, 7, 8, 9]

julia> get!(lru, 2, 1:11)
1:11

julia> lru
LRU{Int64, Vector{Int64}}(; maxsize = 10)
```
Since `1:11` is too big, I would prefer we just skip the cache here entirely, rather than evict any entries and end up with an empty cache.

### PR DESCRIPTION
**Pull request by ericphanson**
Closes https://github.com/JuliaCollections/LRUCache.jl/issues/37

### CODE REVIEW

File: src/LRUCache.jl [hunk 1]
```diff
@@ -146,11 +146,14 @@ function Base.getindex(lru::LRU, key)
...
 function _unsafe_addindex!(lru::LRU{K}, v, key) where K
+    s = lru.by(v)::Int
+    # If entry is larger than entire cache, don't add it
+    s > lru.maxsize && return
     n = LinkedNode{K}(key)
     rotate!(_push!(lru.keyset, n))
-    s = lru.by(v)::Int
     lru.currentsize += s
     lru.dict[key] = (v, n, s)
+    return
```
Comment (ericphanson) [inline src/LRUCache.jl:156]: added this so the return type is `nothing` independently of which branch we take. Note the return of `_unsafe_index!` isn't used anywhere in the codebase

File: Project.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 name = "LRUCache"
 uuid = "8ac3fa9e-de4c-5943-b1dc-09c6b5f20637"
-version = "1.4.1"
+version = "1.5.0"
```
Comment (ericphanson) [inline Project.toml:3]: counted this as a "feature", but I think it could also be seen as a bugfix (and thus a patch release)

**Comment by ericphanson** [comment]
Could you review @jutho?

**Review by Jutho — APPROVED** [review]
Thanks for this very nice PR (and for the review reminder). Looks good to me. I will check whether the GitHub actions need any updating before tagging a the new version.

**Comment by Jutho** [comment]
1.5.0 is tagged, but it seems there is something wrong with tagbot configuration as new tags are not being registered on github. Even 1.4 was also missed as it only goes up to 1.3. Something to look into.

**Comment by ericphanson** [comment]
I see (https://github.com/JuliaCollections/LRUCache.jl/actions/runs/6195513321/job/16820375014#step:3:41):

> Processing version v1.5.0 (ffc929ff9c6b9f98d0e7d9eee8949bd0115a00a6)
> [41](https://github.com/JuliaCollections/LRUCache.jl/actions/runs/6195513321/job/16820375014#step:3:42)
> Generating changelog for version v1.5.0 (ffc929ff9c6b9f98d0e7d9eee8949bd0115a00a6)
> [42](https://github.com/JuliaCollections/LRUCache.jl/actions/runs/6195513321/job/16820375014#step:3:43)
> git@github.com: Permission denied (publickey).
> [43](https://github.com/JuliaCollections/LRUCache.jl/actions/runs/6195513321/job/16820375014#step:3:44)
> fatal: Could not read from remote repository.
> [44](https://github.com/JuliaCollections/LRUCache.jl/actions/runs/6195513321/job/16820375014#step:3:45)
> 
> [45](https://github.com/JuliaCollections/LRUCache.jl/actions/runs/6195513321/job/16820375014#step:3:46)
> Please make sure you have the correct access rights
> [46](https://github.com/JuliaCollections/LRUCache.jl/actions/runs/6195513321/job/16820375014#step:3:47)
> and the repository exists.

maybe the configuration or a key needs to be updated?

**Pull request merged**